### ISSUE TITLE
Test " test.test_manager.test_focus_stays_on_layout_switch" Failing

### ISSUE THREAD

**Issue by aniruddhkanojia** [qtile/qtile#634]
I ran tests after fetching from the latest [commit](https://github.com/qtile/qtile/commit/44d93541551f8a30faab946608adb637e118c8ad) and found this test failing.

```python
======================================================================
ERROR: test.test_manager.test_focus_stays_on_layout_switch
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/samfisher/gsoc/qtile/test/utils.py", line 119, in wrapped_fun
    return function(self)
  File "/home/samfisher/gsoc/qtile/test/test_manager.py", line 733, in test_focus_stays_on_layout_switch
    xephyr.c.nextlayout()
  File "/home/samfisher/gsoc/qtile/libqtile/command.py", line 113, in __call__
    return self.call(self.selectors, self.name, *args, **kwargs)
  File "/home/samfisher/gsoc/qtile/libqtile/command.py", line 246, in call
    raise CommandError(val)
CommandError: No such command.
-------------------- >> begin captured stdout << ---------------------
------------------------ >> begin log file << ------------------------
2015-03-22 20:08:52,757 WARNING init_log:95 Starting Qtile
2015-03-22 20:08:52,826 INFO fire:270 Internal event: addgroup((<libqtile.manager.Qtile object at 0x7f5753b19390>, 'a'), {})
2015-03-22 20:08:52,827 INFO fire:270 Internal event: changegroup((), {})
2015-03-22 20:08:52,828 INFO fire:270 Internal event: addgroup((<libqtile.manager.Qtile object at 0x7f5753b19390>, 'b'), {})
2015-03-22 20:08:52,829 INFO fire:270 Internal event: changegroup((), {})
2015-03-22 20:08:52,830 INFO fire:270 Internal event: addgroup((<libqtile.manager.Qtile object at 0x7f5753b19390>, 'c'), {})
2015-03-22 20:08:52,831 INFO fire:270 Internal event: changegroup((), {})
2015-03-22 20:08:52,833 INFO fire:270 Internal event: addgroup((<libqtile.manager.Qtile object at 0x7f5753b19390>, 'd'), {})
2015-03-22 20:08:52,834 INFO fire:270 Internal event: changegroup((), {})
2015-03-22 20:08:52,837 INFO fire:270 Internal event: setgroup((), {})
2015-03-22 20:08:52,838 INFO fire:270 Internal event: focus_change((), {})
2015-03-22 20:08:52,840 INFO fire:270 Internal event: layout_change((<libqtile.layout.stack.Stack object at 0x7f5753b09a50>, <libqtile.group._Group object at 0x7f5753b098d0>), {})
2015-03-22 20:08:52,896 INFO _xpoll:656 Handling: ScreenChangeNotify
2015-03-22 20:08:52,896 INFO fire:270 Internal event: screen_change((<libqtile.manager.Qtile object at 0x7f5753b19390>, <xcffib.randr.ScreenChangeNotifyEvent object at 0x7f575960b390>), {})
2015-03-22 20:08:52,897 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:52,897 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:52,898 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:52,899 INFO _xpoll:656 Handling: Expose
2015-03-22 20:08:52,899 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:52,900 INFO fire:270 Internal event: startup_once((), {})
2015-03-22 20:08:52,901 INFO fire:270 Internal event: startup((), {})
2015-03-22 20:08:52,903 INFO start:220 Starting server
2015-03-22 20:08:52,910 INFO loop:696 Adding io watch
2015-03-22 20:08:52,911 INFO _xpoll:656 Handling: SelectionNotify
2015-03-22 20:08:52,912 INFO _xpoll:656 Handling: SelectionNotify
2015-03-22 20:08:52,957 INFO connection_made:162 Connection made to server
2015-03-22 20:08:52,958 INFO data_received:166 Data recieved by server
2015-03-22 20:08:52,959 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:52,961 INFO call:91 Command: status((), {})
2015-03-22 20:08:52,962 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:52,963 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:52,964 INFO connection_made:162 Connection made to server
2015-03-22 20:08:52,965 INFO data_received:166 Data recieved by server
2015-03-22 20:08:52,965 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:52,966 INFO call:91 Command: windows((), {})
2015-03-22 20:08:52,966 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:52,967 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:52,970 INFO connection_made:162 Connection made to server
2015-03-22 20:08:52,971 INFO data_received:166 Data recieved by server
2015-03-22 20:08:52,972 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:52,972 INFO call:91 Command: windows((), {})
2015-03-22 20:08:52,973 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:52,973 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:53,062 INFO _xpoll:656 Handling: ConfigureRequest
2015-03-22 20:08:53,063 INFO _xpoll:656 Handling: MapRequest
2015-03-22 20:08:53,068 INFO fire:270 Internal event: window_name_change((), {})
2015-03-22 20:08:53,070 INFO fire:270 Internal event: client_new((Window(one),), {})
2015-03-22 20:08:53,071 INFO fire:270 Internal event: setgroup((), {})
2015-03-22 20:08:53,073 INFO fire:270 Internal event: group_window_add((), {})
2015-03-22 20:08:53,079 INFO fire:270 Internal event: client_focus((Window(one),), {})
2015-03-22 20:08:53,080 INFO fire:270 Internal event: focus_change((), {})
2015-03-22 20:08:53,082 INFO fire:270 Internal event: client_focus((Window(one),), {})
2015-03-22 20:08:53,083 INFO fire:270 Internal event: client_managed((Window(one),), {})
2015-03-22 20:08:53,084 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,085 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,086 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,087 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,088 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,089 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,089 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,090 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,091 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,096 INFO connection_made:162 Connection made to server
2015-03-22 20:08:53,096 INFO data_received:166 Data recieved by server
2015-03-22 20:08:53,097 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:53,097 INFO call:91 Command: windows((), {})
2015-03-22 20:08:53,098 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:53,098 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:53,100 INFO connection_made:162 Connection made to server
2015-03-22 20:08:53,100 INFO data_received:166 Data recieved by server
2015-03-22 20:08:53,101 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:53,101 INFO call:91 Command: windows((), {})
2015-03-22 20:08:53,102 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:53,102 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:53,106 INFO connection_made:162 Connection made to server
2015-03-22 20:08:53,107 INFO data_received:166 Data recieved by server
2015-03-22 20:08:53,109 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:53,110 INFO call:91 Command: windows((), {})
2015-03-22 20:08:53,112 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:53,114 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:53,203 INFO _xpoll:656 Handling: ConfigureRequest
2015-03-22 20:08:53,205 INFO _xpoll:656 Handling: MapRequest
2015-03-22 20:08:53,208 INFO fire:270 Internal event: window_name_change((), {})
2015-03-22 20:08:53,211 INFO fire:270 Internal event: client_new((Window(two),), {})
2015-03-22 20:08:53,212 INFO fire:270 Internal event: setgroup((), {})
2015-03-22 20:08:53,214 INFO fire:270 Internal event: group_window_add((), {})
2015-03-22 20:08:53,219 INFO fire:270 Internal event: client_focus((Window(two),), {})
2015-03-22 20:08:53,220 INFO fire:270 Internal event: focus_change((), {})
2015-03-22 20:08:53,223 INFO fire:270 Internal event: client_focus((Window(two),), {})
2015-03-22 20:08:53,225 INFO fire:270 Internal event: client_managed((Window(two),), {})
2015-03-22 20:08:53,226 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,226 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,227 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,228 INFO _xpoll:656 Handling: UnmapNotify
2015-03-22 20:08:53,228 INFO _xpoll:656 Handling: EnterNotify
2015-03-22 20:08:53,229 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,229 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,230 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,231 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,231 INFO _xpoll:656 Handling: ConfigureNotify
2015-03-22 20:08:53,232 INFO _xpoll:656 Handling: PropertyNotify
2015-03-22 20:08:53,235 INFO connection_made:162 Connection made to server
2015-03-22 20:08:53,236 INFO data_received:166 Data recieved by server
2015-03-22 20:08:53,236 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:53,237 INFO call:91 Command: windows((), {})
2015-03-22 20:08:53,237 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:53,238 INFO eof_received:188 Closing connection on receive EOF
2015-03-22 20:08:53,239 INFO connection_made:162 Connection made to server
2015-03-22 20:08:53,239 INFO data_received:166 Data recieved by server
2015-03-22 20:08:53,240 INFO eof_received:170 EOF recieved by server
2015-03-22 20:08:53,241 INFO eof_received:186 Sending result on receive EOF
2015-03-22 20:08:53,241 INFO eof_received:188 Closing connection on receive EOF
------------------------- >> end log file << -------------------------

--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 174 tests in 85.965s

FAILED (SKIP=2, errors=1)
```

**Comment by aniruddhkanojia** [comment]
I think it's because of use of [outdated methods](https://github.com/qtile/qtile/blob/develop/test/test_manager.py#L728) which were removed by a045bbc72bf66f6af3977625f04bdd913dec6a41

**Comment by aniruddhkanojia** [comment]
Fixed after 9d3df78266ca9fb4d098fab1794f5dedc82be19c
